src/php/Printer/Printer.php

```diff
@@ -53,49 +54,64 @@ private function __construct(bool $readable)
      */
     public function print($form): string
     {
-        $printerName = $this->getTypePrinterName($form);
-        $printer = $this->createTypePrinterByName($printerName);
+        $printer = $this->createTypePrinter($form);
 
         return $printer->print($form);
     }
 
     /**
-     * @param mixed $form
+     * @param mixed $form The form to print
      */
-    private function getTypePrinterName($form): string
+    private function createTypePrinter($form): TypePrinterInterface
     {
-        return 'object' === gettype($form)
-            ? get_class($form)
-            : gettype($form);
+        if (is_object($form)) {
+            return $this->createObjectTypePrinter($form);
+        }
+
+        return $this->creatScalarTypePrinter($form);
     }
 
-    private function createTypePrinterByName(string $printerName): TypePrinterInterface
+    /**
+     * @param mixed $form The form to print
+     */
+    private function createObjectTypePrinter($form): TypePrinterInterface
     {
-        if (Tuple::class === $printerName) {
+        if ($form instanceof Tuple) {
             return new TuplePrinter($this);
         }
-        if (Keyword::class === $printerName) {
+        if ($form instanceof Keyword) {
             return new KeywordPrinter();
         }
-        if (Symbol::class === $printerName) {
+        if ($form instanceof Symbol) {
             return new SymbolPrinter();
         }
-        if (Set::class === $printerName) {
+        if ($form instanceof Set) {
             return new SetPrinter($this);
         }
-        if (PhelArray::class === $printerName) {
+        if ($form instanceof PhelArray) {
             return new PhelArrayPrinter($this);
         }
-        if (AbstractStruct::class === $printerName) {
-            return new StructPrinter($this);
-        }
-        if (Table::class === $printerName) {
+        if ($form instanceof Table) {
             return new TablePrinter($this);
         }
+        if ($form instanceof AbstractStruct) {
```

Comment:
I think you need to test that before the `Table` since `AbstractStruct` extends `Table`

Reply:
That's why `Table` condition is before and `AbstractStruct` is after.

Reply:
Yes you are correct!